src/client/client4.js

```diff
@@ -66,6 +70,17 @@ export default class Client4 {
         this.token = token;
     }
 
+    setCSRFFromCookie() {
+        const cookies = document.cookie.split(';');
```

Comment:
I don't know if we can assume we'll have the cookie here depending on what environment this is running on. I don't know if this is needed outside of the web though, so maybe we just add a null check for `document.cookie` before doing this?

Reply:
Good call, yeah. On other environments token-based authentication is used, I suppose? Is the reason the tests fail with `document.cookie` undefined because the direct access to `document.cookie` isn't supposed to happen with Redux or is that rather a mocking problem, since it works fine in the application itself?

Reply:
I haven't thought too much about whether or not we just shouldn't use `document.cookie` in this code. If we're really trying to make this code platform-agnostic, we wouldn't access it at all since it only exists on the browser and not on mobile/node.js, but I don't think it's worth that extra effort when we could just null check it. As for the tests, that's just an example of an environment without `document.cookie`, so while mocking it would fix the tests, it doesn't fix the root problem.

Reply:
Absolutely, makes sense. Updated the code to check if `document` and `document.cookie` are defined.